vault/core.go

```diff
@@ -1273,6 +1273,15 @@ func (c *Core) unsealPart(ctx context.Context, seal Seal, key []byte, useRecover
 				return nil, errors.New("did not get expected recovery information to set new seal during migration")
 			}
 
+			if err := c.seal.SetBarrierConfig(ctx, &SealConfig{
+				Type:            wrapping.Shamir,
+				SecretShares:    1,
+				SecretThreshold: 1,
+				StoredShares:    1,
```

Comment:
I think you may want something along the lines of:

```
				Type:            wrapping.Shamir,
				SecretShares:    recoveryConfig.SecretShares,
				SecretThreshold: recoveryConfig.SecretThreshold,
				StoredShares:    1,
```

Reply:
Done. Thanks to @ncabatoff!